java/fury-core/src/main/java/org/apache/fury/serializer/collection/CollectionSerializers.java

```diff
@@ -411,6 +415,66 @@ public ConcurrentSkipListSet newCollection(MemoryBuffer buffer) {
     }
   }
 
+  public static final class SetFromMapSerializer extends CollectionSerializer<Set<?>> {
+
+    private static final long MAP_FIELD_OFFSET;
+
+    static {
+      try {
+        Field mapField = Class.forName("java.util.Collections$SetFromMap").getDeclaredField("m");
+        MAP_FIELD_OFFSET = Platform.objectFieldOffset(mapField);
+      } catch (final Exception e) {
+        throw new RuntimeException(e);
+      }
+    }
+
+    public SetFromMapSerializer(Fury fury, Class<Set<?>> type) {
+      super(fury, type, false);
+    }
+
+    @Override
+    public Collection newCollection(MemoryBuffer buffer) {
+      int numElements = buffer.readVarUint32Small7();
+      setNumElements(numElements);
+      final ClassInfo mapClassInfo = fury.getClassResolver().readClassInfo(buffer);
+      final MethodHandle methodHandle = ReflectionUtils.getCtrHandle(mapClassInfo.getCls(), true);
```

Comment:
This won't work for graalvm. It's too late, we should move it to constructor, so we can generate code at graalvm  build time

Reply:
ok, I change it. I test this version in graalvm and it works, I wandering why.

Reply:
Could you add a test in graalvm_tests module?